### PR TITLE
Ported @zapjs/curve and its types;

### PR DESCRIPTION
**Pull request by acemasterjb**
Should note that python does now have switch statements, a dict with lambda values was used instead;

BigNumbers are integers by default in python;

Curve/regex holds regex pattern for Curve/curve

---

Tackles issue #8 

needs to be mock tested

### CODE REVIEW

File: Curve/types.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from typing import NewType, TypedDict
+
+CurveType = NewType("CurveType", list)
+
+
+class CureveTerm(TypedDict):
+    fn: int or float
+    power: int or float
```
Comment (superposition) [inline Curve/types.py:8]: please include tests before submission

**Review by superposition — CHANGES_REQUESTED** [review]

**Comment by acemasterjb** [comment]
@andrew-fleming can you help me out with this change?

**Comment by andrew-fleming** [comment]
@acemasterjb yeah, I'll take a look

**Comment by acemasterjb** [comment]
Curve\test\Curve_test: Added testing via pytests; Tested 1 case for checkValidity 

will add more tests soon.

**Comment by acemasterjb** [comment]
This pull also removes the cloned web3

**Comment by superposition** [comment]
hey resolve these conflicts!

**Pull request merged**